### PR TITLE
RestAPI annotation of factoryinsight

### PR DESCRIPTION
**Pull request by Sphingobium**
# Description

openAPI 2.0 conform annotation of the restAPI endpoints for automatic generation of the documentation. Also removes the dysfunctional endpoint "factorylocations"

Fixes # (1210)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

only documentation

# Checklist:

- [X] I have made corresponding changes to the documentation
- [ ] I have made corresponding changes to the changelog (if needed).
- [ ] I have made corresponding notices to the upgrade instructions (if needed)

# Changelog changes

I am not sure if this requires one

### CODE REVIEW

**Review by Scarjit — CHANGES_REQUESTED** [review]
Use [// @securityDefinitions.basic BasicAuth](https://github.com/swaggo/swag/blob/master/README.md#security) for all protected endpoints

File: golang/cmd/factoryinsight/http.go [hunk 1]
```diff
@@ -44,6 +50,7 @@ func SetupRestAPI(accounts gin.Accounts, version string) {
 		v1.GET("/:customer/:location/:asset", getValuesHandler)
 		v1.GET("/:customer/:location/:asset/:value", getDataHandler)
 	}
+	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
```
Comment (JeremyTheocharis) [inline golang/cmd/factoryinsight/http.go:53]: There should only be documentation and no changes in the source code

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: golang/cmd/factoryinsight/http.go (hunk 1 above)
Comment (Sphingobium) [reply golang/cmd/factoryinsight/http.go]: Removed line and imports

**Comment by Sphingobium** [comment]
Requested changes are added

**Review by Scarjit — APPROVED** [review]

**Comment by Sphingobium** [comment]
Shelve documentation due to explosion of scope - should this still be merged into main?

**Comment by Scarjit** [comment]
@JeremyTheocharis I would still merge this.

**Review by Scarjit — APPROVED** [review]

**Pull request closed without merging**